subsys/bluetooth/mesh/sar_cfg_srv.c

```diff
@@ -135,6 +159,41 @@ static int sar_cfg_srv_init(struct bt_mesh_model *model)
 	return 0;
 }
 
+static void sar_cfg_srv_reset(struct bt_mesh_model *model)
+{
+	struct bt_mesh_sar_tx sar_tx = BT_MESH_SAR_TX_INIT;
+	struct bt_mesh_sar_rx sar_rx = BT_MESH_SAR_RX_INIT;
+
+	memcpy(&bt_mesh.sar_tx, &sar_tx, sizeof(struct bt_mesh_sar_tx));
+	memcpy(&bt_mesh.sar_rx, &sar_rx, sizeof(struct bt_mesh_sar_rx));
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		sar_rx_store(model, true);
+		sar_tx_store(model, true);
+	}
+}
+
+#ifdef CONFIG_BT_SETTINGS
+static int sar_cfg_srv_settings_set(struct bt_mesh_model *model, const char *name, size_t len_rd,
+				    settings_read_cb read_cb, void *cb_data)
+{
+	if (!strncmp(name, "sar_rx", 2)) {
+		(void)read_cb(cb_data, &bt_mesh.sar_rx, sizeof(bt_mesh.sar_rx));
```

Comment:
Should we use `bt_mesh_settings_set` here? It additionally checks for return code of `read_cb` and length of the read data.